### ISSUE TITLE
Scala scripts hanging indefinitely on Mac OS.

### ISSUE THREAD

**Issue by friedbrice** [scala/bug#10744]
Basic Scala scripts are hanging indefinitely without executing when invoked on Mac OS.

Here is a minimal example:

```
#!/bin/sh
exec scala "$0" "$@"
!#

println("hello, world")
```

Executing this script at the command line does not produce any output and does not exit.

Here are the outputs of `scala -version`, `java -version`, and `sw_vers`:

```
$ scala -version
Scala code runner version 2.12.4 -- Copyright 2002-2017, LAMP/EPFL and Lightbend, Inc.
$ java -version
java version "1.8.0_152"
Java(TM) SE Runtime Environment (build 1.8.0_152-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.152-b16, mixed mode)
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.3
BuildVersion:	17D47
```

**Comment by SethTisue** [comment]
does running `fsc -shutdown` first fix it?

and/or, does adding `-nocompdaemon` after `exec scala` fix it?

**Comment by friedbrice** [comment]
`fsc -shutdown` hangs indefinitely.

Adding `-nocompdaemon` after `exec scala` results in a working shell script.

Thanks much!

**Comment by alvinj** [comment]
It looks like this is an old bug, but I just started having the same problem after upgrading to MacOS 10.14.4: old Scala shell scripts that used to work no longer work. But adding `-nocompdaemon` after `exec scala` fixes the problem. Thanks!

**Comment by SethTisue** [comment]
`-nocompdaemon` will be the default starting with Scala 2.13

**Comment by som-snytt** [comment]
As noted, `-nc` fixes it on 2.12 and is the default for 2.13.
